### ISSUE TITLE
Returned promises do not get resolved

### ISSUE THREAD

**Issue by kmc059000** [angular/protractor#1757]
I was previously on 1.3.1 and just upgraded to 1.6.1. The following test worked in 1.3.1:

````javascript
it('should use expect in then()', function() {
    element(by.css('#el')).isDisplayed().then(function(isDisplayed) {
        //do less trivial stuff here
        console.log('i expect this to be printed');
        expect(isDisplayed).toBe(true);
        browser.pause();
    });
});
`````

In 1.3.1, the expectation is evaluated, a message is logged, and protractor pauses. However in 1.6.1, I have not seen any evidence that my callback that is passed into ````then()```` is ever called.

I also tried updating to 1.4.0 instead of 1.6.1, and I received the same issue. I have not upgraded to jasmine 2.

Any ideas what may cause this? Nothing stood out to me in the 

I just tried this, and protractor prints that there is 1 test, and only 1 assertion. which is not correct since there are 3:

````javascript
it('should expect things', function() {
    //these are ignored
    expect(element(by.css('#a')).isDisplayed()).toBe(true);
    expect(element(by.css('#a')).isDisplayed()).toBe(false);

    //this is evaluated
    expect(true).toBe(false);
});
````

Thanks, let me know if there is more information you need

**Comment by hankduan** [comment]
Can you try going back to 1.3.1 to see if everything works still just to remove the possibility that it could be a change in your app? I just tried it on my machine with:

```
describe('test', function() {
  it('should use expect in then()', function() {
    browser.get('#')
      element(by.css('a')).isDisplayed().then(function(isDisplayed) {
          console.log('i expect this to be printed');
          expect(isDisplayed).toBe(true);
          browser.pause();
      });
  });
});
```

```
$ node /tmp/node_modules/.bin/protractor spec/hankConf.js --suite hank
Using the selenium server at http://localhost:4444/wd/hub
[launcher] Running 1 instances of WebDriver
Started
WARNING - more than one element found for locator By.cssSelector("a") - the first result will be used
i expect this to be printed
Hit SIGUSR1 - starting debugger agent.
debugger listening on port 5858
Starting WebDriver debugger in a child process. Pause is still beta, please report issues at github.com/angular/protractor
------- WebDriver Debugger -------
 ready

press c to continue to the next webdriver command
press d to continue to the next debugger statement
type "repl" to enter interactive mode
type "exit" to break out of interactive mode
press ^C to exit

-- WebDriver control flow schedule 
 |- waiting for debugger to attach
 |---    at /workspace/protractor/spec/hank_spec.js:10:19
wd-debug> 
```

**Comment by kmc059000** [comment]
Yes, I just reverted the commit that upgraded protractor, and it works as expected in 1.3.1. I had done this earlier today too, so it isn't a fluke.

I'm pretty sure its something with our repository since I can run the tests in the protractor repository no problem. I'm not sure what though.

**Comment by hankduan** [comment]
What is the behavior: are there error messages, does it just skip `fn` in `.isDisplayed().then(fn)`?
Can you paste what you see in the output?

**Comment by kmc059000** [comment]
That is what it seems like. When chrome opens, it immediately closes. Its like nothing is getting added to the queue or protractor is ignoring the queue.

````
DEBUG - Running with --troubleshoot
DEBUG - Protractor version: 1.6.1
DEBUG - Your base url for tests is http://localhost:9002/modules/common/index.html
DEBUG - Attempting to find the SeleniumServerJar in the default location used by webdriver-manager
DEBUG - Attempting to find the chromedriver binary in the default location used by webdriver-manager
Starting selenium standalone server...
[launcher] Running 1 instances of WebDriver
Selenium standalone server started at http://169.254.80.80:58914/wd/hub
DEBUG - WebDriver session successfully started with capabilities { caps_:
   { applicationCacheEnabled: false,
     rotatable: false,
     mobileEmulationEnabled: false,
     chrome: { userDataDir: 'C:\\Users\\KCRAWF~1\\AppData\\Local\\Temp\\scoped_dir11412_14948' },
     takesHeapSnapshot: true,
     databaseEnabled: false,
     handlesAlerts: true,
     version: '40.0.2214.91',
     platform: 'WIN8_1',
     browserConnectionEnabled: false,
     nativeEvents: true,
     acceptSslCerts: true,
     'webdriver.remote.sessionid': 'd2b0e473-0f19-42fb-947c-cff84af03e0d',
     locationContextEnabled: true,
     webStorageEnabled: true,
     browserName: 'chrome',
     takesScreenshot: true,
     javascriptEnabled: true,
     cssSelectorsEnabled: true } }
Dashboard page spec
  should save collapsed states for tiles even after one is removed - pass


Finished in 0.131 seconds
1 test, 0 assertions, 0 failures

Shutting down selenium standalone server.
````

**Comment by kmc059000** [comment]
I have chrome driver 2.13 and selenium driver 2.44.0.

**Comment by kmc059000** [comment]
````javascript
it('should pause', function() {
    browser.pause();
});
````

That doesnt pause for me either.

**Comment by hankduan** [comment]
Can you clarify if the issue is only with pause or with running protractor in general?

**Comment by kmc059000** [comment]
Running protractor in general. None of the webdriver promises are getting resolved, and protractor does not appear to be executing the commands. (click, getText, isDisplayed, etc)

**Comment by FuturaITLtd** [comment]
This might sound like a noddy suggestion, but have you tried running
manually deleting the protractor directory from node modules, doing a npm
cache clean and fresh npm install to upgrade to the latest protractor. The
reason I suggest this is because a few releases back we encountered some
strange behaviour where by previous protractor functionality seemed broken
and it turned out that the upgrade hadn't quite upgraded!

**Comment by kmc059000** [comment]
@FuturaITLtd I did that and I also uninstalled some packages that were installed at the top level of node_modules\ which protractor depended on. That seems to have fixed the issue. I'm not 100% sure if it was the ````npm cache clean```` or removal of those packages though.

Thanks everybody.

**Comment by kmc059000** [comment]
I believe I narrowed it down. We had jasminewd installed in our root node_modules\ folder. Whenever jasminewd is installed in the root, and I uninstall 1.3.1 and install 1.6.1, I get the issue.

If I uninstall jasminewd from the root node_modules\ and then uninstall protractor 1.6.1, protractor works as expected. I even tried uninstalling jasminewd and reinstalling it and then updating protractor and got the same results.

In short:
jasminewd at root. uninstall protractor, install protractor -> **problems**
jasminewd at root, uninstall jasminewd, install jasminewd, uninstall protractor, install protractor -> **problem**

jasminewd at root, uninstall jasminewd, uninstall protractor, install protractor -> **WORKS**

**Comment by bertjan** [comment]
@kmc059000 same issue here with Protractor 3.x, the workaround you described in the comment above works, I'm guessing that the wrong jasminewd(2) version gets pulled in by another dependency. I'll update here when I know more.
